skxray/cdi.py

```diff
@@ -106,30 +89,8 @@ def gauss(dims, sigma):
     Returns
     -------
     Array :
-        2D or 3D gaussian
-    """
-    x = dist(dims)
-    y = np.exp(-(x / sigma)**2/2.)
-    return y/np.sum(y)
-
-
-def convolution(array1, array2):
-    """
-    Calculate convolution of two arrays. Transfer into q space to perform the calculation.
-
-    Parameters
-    ----------
-    array1 : array
-        The size of array1 needs to be normalized.
-    array2 : array
-        The size of array2 keeps the same
-
-    Returns
-    -------
-    array :
-        convolution result
+        ND gaussian
     """
-    fft_norm = lambda x:  np.fft.fftshift(np.fft.fftn(x)) / np.sqrt(np.size(x))
-    fft_1 = fft_norm(array1)
-    fft_2 = fft_norm(array2)
-    return np.abs(np.fft.ifftshift(np.fft.ifftn(fft_1*fft_2)) * np.sqrt(np.size(array2)))
```

Comment:
I compared with scipy.signal.fftconvolve, and found a bit difference on the boundary part. This is not critical, as we mainly pay attention to the center part other than the boundary in CDI reconstruction. 
I will still keep this function for late use, but keep in mind scipy.signal.fftconvolve is another option.